python/ray/autoscaler/_private/commands.py

```diff
@@ -485,13 +487,15 @@ def monitor_cluster(cluster_config_file: str, num_lines: int,
         port_forward=None)
 
 
-def warn_about_bad_start_command(start_commands: List[str]) -> None:
+def warn_about_bad_start_command(start_commands: List[str],
+                                 no_monitor_on_head: bool = False) -> None:
     ray_start_cmd = list(filter(lambda x: "ray start" in x, start_commands))
     if len(ray_start_cmd) == 0:
         cli_logger.warning(
             "Ray runtime will not be started because `{}` is not in `{}`.",
             cf.bold("ray start"), cf.bold("head_start_ray_commands"))
-    if not any("autoscaling-config" in x for x in ray_start_cmd):
+    if not (any("autoscaling-config" in x
+                for x in ray_start_cmd) or no_monitor_on_head):
```

Comment:
Just wanted to clarify the expected logic for this line in the K8s operator use case where I assume `no_monitor_on_head`=`true`:

The warning will be logged.  Is that the correct/intended interpretation?

Reply:
If no_monitor_on_head is true, the if condition becomes
not (A or True) = False,
so the warning isn’t logged.

The readability here could be improved.

Reply:
Got it.  And I see you improved read ability as well in the PR update.  Thank you!